e2e/validators.go

```diff
@@ -430,7 +430,12 @@ func GetFieldFromJsonObjectOnNode(ctx context.Context, s *Scenario, fileName str
 
 // ValidateTaints checks if the node has the expected taints that are set in the kubelet config with --register-with-taints flag
 func ValidateTaints(ctx context.Context, s *Scenario) {
-	expectedTaints := s.Runtime.NBC.KubeletConfig["--register-with-taints"]
+	var expectedTaints string
+	if s.Runtime.NBC != nil {
+		expectedTaints = s.Runtime.NBC.KubeletConfig["--register-with-taints"]
+	} else {
+		expectedTaints = s.Runtime.AKSNodeConfig.KubeletConfig.KubeletFlags["--register-with-taints"]
```

Comment:
Can this throw panic?

Additionally, I'd prefer to pass the expected taints explicitly from the scenario validator.

Reply:
My previous version was passing the kubeletconfig. But then I found that it's part of the `scenario` so updated with this version. Since this is just a test, I think I can change it to passing the kubeletconfig directly for explicitly.

Reply:
I would change the function signature to something like this:
`func ValidateTaints(ctx context.Context, s *Scenario, expectedTaints string)`

Reply:
agreed that's even better